src/log_batch.rs

```diff
@@ -497,6 +503,7 @@ impl LogItemBatch {
         }
         let checksum = crc32(&buf[offset..]);
         buf.encode_u32_le(checksum)?;
```

Comment:
Don't encode, leave a placeholder here, and comment it's delayed to prepare_write.